op-node/rollup/derive/channel_mux_test.go

```diff
@@ -0,0 +1,69 @@
+package derive
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-node/metrics"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func TestChannelMux_LaterHolocene(t *testing.T) {
+	log := testlog.Logger(t, log.LevelTrace)
+	ctx := context.Background()
+	l1A := eth.L1BlockRef{Time: 0, Hash: common.Hash{0xaa}}
+	l1B := eth.L1BlockRef{Time: 12, Hash: common.Hash{0xbb}}
+	cfg := &rollup.Config{
+		HoloceneTime: &l1B.Time,
+	}
+	spec := rollup.NewChainSpec(cfg)
+	m := metrics.NoopMetrics
+	c := NewChannelMux(log, spec, nil, m)
+
+	require.IsType(t, new(ChannelBank), c.RawChannelProvider)
+
+	err := c.Reset(ctx, l1A, eth.SystemConfig{})
+	require.Equal(t, io.EOF, err)
+	require.IsType(t, new(ChannelBank), c.RawChannelProvider)
+
+	c.Transform(rollup.Holocene)
```

Comment:
Probably worth a test (possibly a separate method) where Reset is called for a post-holocene block without having previously called `c.Transform` (ie the case where we startup with existing data that is beyond holocene activation.

Reply:
I think the existing `TestChannelMux_GenesisHolocene` is already covering this case. Just the test name itself is misleading. It's not really about Holocene being active at genesis, but active at the L1 block we call reset on. Will update the test to clarify this.